crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -1 +1,200 @@
+use std::sync::Arc;
+
+use axum::{
+    extract::{Json, State},
+    Router,
+};
+use revm::db::components::state::StateRef;
+use tokio::sync::Mutex;
+
+use rethnet_eth::{
+    remote::{client::Request as RpcRequest, jsonrpc, MethodInvocation},
+    U256,
+};
+use rethnet_evm::state::{HybridState, RethnetLayer};
+
 mod methods;
+
+type StateType = Arc<Mutex<HybridState<RethnetLayer>>>;
+
+pub async fn router(state: StateType) -> Router {
```

Comment:
I recommend following the pattern used in the readme of axum: https://crates.io/crates/axum/

You can add types to the function signature that tell it what type of Json object you're expecting, reducing boilerplate code

![image](https://github.com/NomicFoundation/hardhat/assets/6917585/9839f971-21cb-4dde-a963-074b338b15fb)

Reply:
i could use some more input on what exactly you'd like to see happen here. my understanding is that we'll only have a single "handler", because we'll only ever route to one "endpoint": `/`. so, i'm not sure what benefit there would be to breaking it out into a separate function like this. i did make a small change to have `axum` do the JSON parsing instead of doing it ourselves. see that [here](https://github.com/NomicFoundation/hardhat/pull/3998/commits/afc1a6d0be2cc29bf3303d6ba27568371053856b). is that what you had in mind?

Reply:
Yes, we'll have a single handler, but by using the code as described in the example, it automatically handles and responds with an error for the failure case when parsing. It reduces boilerplate for you.

Reply:
i see, but i'm still not seeing how you'd like to see it applied to the code here, especially now that i've introduced all the `handle_*` functions. if you think there's still room for further improvement beyond the introduction of those functions, let's discuss on a call.